spec/controllers/sufia/admin/permission_templates_controller_spec.rb

```diff
@@ -40,5 +40,24 @@
         expect(flash[:notice]).to eq 'Permissions updated'
       end
     end
+    describe "update visibility" do
+      let(:admin_set) { create(:admin_set) }
+      let!(:permission_template) { Sufia::PermissionTemplate.create!(admin_set_id: admin_set.id) }
+      let(:input_params) do
+        { admin_set_id: admin_set.id,
+          sufia_permission_template: {
+            visibility: 'open'
+          } }
+      end
+
+      it "is successful" do
+        expect(controller).to receive(:authorize!).with(:update, permission_template)
+        expect do
+          put :update, params: input_params
+        end.to change { permission_template.visibility }.from(nil).to('open')
```

Comment:
This because your permission_template is loaded into memory just once. It's not aware that the backing data has been changed.  You could do `permission_template.reload.visibility` and it'll probably fix it.